### PR TITLE
Downgrade ambiguous precedence error to warning

### PR DESCRIPTION
**Pull request by gw000**
PR checklist:

- [x] Added changelog entry.

### CODE REVIEW

File: polar-core/tests/integration_tests.rs [hunk 1]
```diff
@@ -1560,20 +1560,22 @@ fn test_and_or_warning() -> TestResult {
     p.load_str("f(x) if (\n\t    x = 1 and  x > 1) or x < 3;")?;
 
     p.clear_rules();
+    p.load_str("f(x) if x = 1 and x > 1 or x < 3;")?;
+    let msg = p.next_message().unwrap();
+    assert!(matches!(&msg.kind, MessageKind::Warning));
+    assert_eq!(
+        &msg.msg,
+        "Expression without parentheses could be ambiguous. \nPrior to 0.20, `x and y or z` would parse as `x and (y or z)`. \nThis was changed in 0.20 to match other languages. \n\n\n001: f(x) if x = 1 and x > 1 or x < 3;\n             ^"
```
Comment (gj) [inline polar-core/tests/integration_tests.rs:1568]: nit on the warning message
```suggestion
        "Expression without parentheses could be ambiguous. \nPrior to 0.20, `x and y or z` would parse as `x and (y or z)`. \nAs of 0.20, it parses as `(x and y) or z`, matching other languages. \n\n\n001: f(x) if x = 1 and x > 1 or x < 3;\n             ^"
```

Comment (gw000) [reply polar-core/tests/integration_tests.rs:1568]: 👍

File: polar-core/src/warnings.rs [hunk 2]
```diff
@@ -159,19 +159,13 @@ impl<'kb> AndOrPrecendenceCheck<'kb> {
             .map(|(source, or_term)| {
                 let mut msg = "Expression without parentheses could be ambiguous. \n\
                     Prior to 0.20, `x and y or z` would parse as `x and (y or z)`. \n\
-                    This was changed in 0.20 to match other languages. \n\
+                    As of 0.20, it parses as `(x and y) or z`, matching other languages \n\
```
Comment (gj) [inline polar-core/src/warnings.rs:162]:
```suggestion
                    As of 0.20, it parses as `(x and y) or z`, matching other languages. \n\
```

File: polar-core/tests/integration_tests.rs [hunk 3]
```diff
@@ -1560,20 +1560,22 @@ fn test_and_or_warning() -> TestResult {
     p.load_str("f(x) if (\n\t    x = 1 and  x > 1) or x < 3;")?;
 
     p.clear_rules();
+    p.load_str("f(x) if x = 1 and x > 1 or x < 3;")?;
+    let msg = p.next_message().unwrap();
+    assert!(matches!(&msg.kind, MessageKind::Warning));
+    assert_eq!(
+        &msg.msg,
+        "Expression without parentheses could be ambiguous. \nPrior to 0.20, `x and y or z` would parse as `x and (y or z)`. \nAs of 0.20, it parses as `(x and y) or z`, matching other languages \n\n\n001: f(x) if x = 1 and x > 1 or x < 3;\n             ^"
```
Comment (gj) [inline polar-core/tests/integration_tests.rs:1568]:
```suggestion
        "Expression without parentheses could be ambiguous. \nPrior to 0.20, `x and y or z` would parse as `x and (y or z)`. \nAs of 0.20, it parses as `(x and y) or z`, matching other languages. \n\n\n001: f(x) if x = 1 and x > 1 or x < 3;\n             ^"
```

File: polar-core/tests/integration_tests.rs [hunk 4]
```diff
@@ -1560,20 +1560,22 @@ fn test_and_or_warning() -> TestResult {
...
-            .kind,
-        ErrorKind::Parse(ParseError::AmbiguousAndOr { .. })
-    ));
+    p.clear_rules();
+    p.load_str("f(x) if x = 1 or x > 1 and x < 3;")?;
+    let msg = p.next_message().unwrap();
+    assert!(matches!(&msg.kind, MessageKind::Warning));
+    assert_eq!(
+        &msg.msg,
+        "Expression without parentheses could be ambiguous. \nPrior to 0.20, `x and y or z` would parse as `x and (y or z)`. \nAs of 0.20, it parses as `(x and y) or z`, matching other languages \n\n\n001: f(x) if x = 1 or x > 1 and x < 3;\n                      ^"
```
Comment (gj) [inline polar-core/tests/integration_tests.rs:1577]:
```suggestion
        "Expression without parentheses could be ambiguous. \nPrior to 0.20, `x and y or z` would parse as `x and (y or z)`. \nAs of 0.20, it parses as `(x and y) or z`, matching other languages. \n\n\n001: f(x) if x = 1 or x > 1 and x < 3;\n                      ^"
```

**Review by gj — APPROVED** [review]

**Pull request merged**